src/txmempool.cpp

```diff
@@ -139,13 +167,13 @@ void CTxMemPool::UpdateTransactionsFromBlock(const std::vector<uint256> &vHashes
                 assert(childIter != mapTx.end());
                 // We can skip updating entries we've encountered before or that
                 // are in the block (which are already accounted for).
-                if (!visited(childIter) && !setAlreadyIncluded.count(childHash)) {
+                if (!visited(childIter) && !mapMemPoolDescendantsToUpdate.count(childIter)) {
```

Comment:
Not sure if this PR is still active, but I believe it may be more efficient to check:
```diff

+++ if (!visited(childIter) && !childIter->GetMemPoolParentsConst().count(it) {
```
since if the child is unvisited and is in `vHashesToUpdate`, it will have `it` as a parent, which isn't the case otherwise. I think this would be more efficient if the childiter's set of parents is < cache size, which I would expect to be true more often than not